Allow multiple email opt-ins. Add interest group tab.

katzmo opened the pull request.

katzmo committed:
[form builder] allow multiple email opt-ins

katzmo committed:
[newsletters] Refactor Subscription → QueueItem

Send unsubscribes on delete even if subscription is new.

katzmo committed:
[newsletters] Refactor Component → Subscription

katzmo committed:
[newsletters] Merge subscriptions (part I)

Before saving, merge subscriptions from the same submission to reduce
queue items and make sure opt-ins will win over opt-outs of the same list.

katzmo committed:
[newsletters] Merge subscriptions (part II)

Merge the data of 2 subscriptions.

torotil reviewed (changes requested):
Ok, that breaks mailchimp interest groups at the moment.

torotil committed:
Simplify subscription merging. Introduce $subscription->components.

torotil committed:
Remove optin_statement recording from newsletter activities.

campaignion_opt_in does a far better job at tracking opt-ins at this
point. Subscription activities can now focus on actual subscription
changes.

torotil committed:
Add missing dependency on libraries.

torotil committed:
Add #id for easier #states usage in other modules.

torotil committed:
Implement groups tab.

- hook into webform_component_edit_form to add the fields.
- Use #states to hide groups for not selected lists.
- Integrate with form_builder.
- Add lists in hook_campaignion_newsletters_mailchimp_data_alter().

torotil committed:
Add test for campaignion_newsletters_mailchimp_interests.

torotil committed:
Refactor merge(). Don’t pass flags through the subscription object.

- Subscription::merge() now takes into account the ->delete flag.
- Removed optin_info, send_welcome and send_optin from the subscription
  object. They are now calculated on-the-fly when the QueueItem is
  created.

torotil committed:
Refactor Component::unsubscribe().

torotil committed:
Componen::getSubscriptions() should always return an array.
